lib/drivers/medtronic600/medtronic600Simulator.js

```diff
@@ -482,16 +482,29 @@ class Medtronic600Simulator {
 
       if (suppressedBasal != null) {
         if (Date.parse(suppressedBasal.basalEndTime) > Date.parse(activeSuspendedBasal.time)) {
-          suppressedBasal.duration = Date.parse(activeSuspendedBasal.time) -
-            Date.parse(suppressedBasal.time);
-          // It's possible for duration to get down to 0 here (which is fine).
-          if (suppressedBasal.duration < 0) {
-            throw new Error('Should never get a duration of less than zero');
-          } else if (suppressedBasal.duration === 0) {
+          if (Date.parse(activeSuspendedBasal.time) >= Date.parse(suppressedBasal.time)) {
+            suppressedBasal.duration = Date.parse(activeSuspendedBasal.time) -
+              Date.parse(suppressedBasal.time);
+          } else {
+            // The suspend event is the first basal event in the history.
+            // Because we don't know what it's suspending, match it to the beginning of the
+            // first suspended event.
+            activeSuspendedBasal.time = suppressedBasal.time;
+            activeSuspendedBasal.deviceTime = suppressedBasal.deviceTime;
+            activeSuspendedBasal.duration =
+              Date.parse(activeSuspendedBasal.basalEndTime).valueOf() -
+              Date.parse(suppressedBasal.time).valueOf();
             this.removeEvent(suppressedBasal);
           }
         }
 
+        // It's possible for suppressed basal duration to get down to 0 here (which is fine).
+        if (suppressedBasal.duration < 0) {
+          throw new Error('Should never get a duration of less than zero');
+        } else if (suppressedBasal.duration === 0) {
+          this.removeEvent(suppressedBasal);
```

Comment:
Why do we need to remove this event? It's not uncommon to have basals with zero duration.

Reply:
In this function (`applySuspendedBasals`), we're synthesizing (in an overlay fashion) suspended basals "over" the existing schedule (scheduled, temp and auto basals). So they're not basals that occured with 0 duration in the event history.